From: jiyun_ha <jiyun_ha@tmax.co.kr>
Date: Fri, 29 Oct 2021 17:37:12 +0900
Subject: [feat] osk-hangul : compile scss file when building the project

 - Previously, the css file is installed when installing this project.
   Since build fails if stylesheet.css not exists, added a dummy file to
   avoid the problem. Instead of `custom_target`, using `run_command`
   will solve the problem, since it runs the command at build time.

[IMS] N/A
---
 extensions/meson.build               | 4 +++-
 extensions/osk-hangul/meson.build    | 9 +--------
 extensions/osk-hangul/stylesheet.css | 1 -
 3 files changed, 4 insertions(+), 10 deletions(-)
 delete mode 100644 extensions/osk-hangul/stylesheet.css

diff --git a/extensions/meson.build b/extensions/meson.build
index ca00d01..54a49d5 100644
--- a/extensions/meson.build
+++ b/extensions/meson.build
@@ -14,10 +14,12 @@ foreach e : enabled_extensions
   metadata_conf.set('shell_current', shell_version)
   metadata_conf.set('url', 'https://gitlab.gnome.org/GNOME/gnome-shell-extensions')
 
+
   extension_sources = files(e + '/extension.js')
-  extension_data = files(e + '/stylesheet.css')
+  extension_data = []
 
   subdir(e)
+  extension_data += files(e + '/stylesheet.css')
 
   install_data (extension_sources + extension_data,
     install_dir: join_paths(extensiondir, uuid)
diff --git a/extensions/osk-hangul/meson.build b/extensions/osk-hangul/meson.build
index e1ed2bf..9e40d86 100644
--- a/extensions/osk-hangul/meson.build
+++ b/extensions/osk-hangul/meson.build
@@ -1,12 +1,5 @@
 sassc = find_program('sassc')
-style = custom_target('stylesheet.css',
-                      input: '_stylesheet.scss',
-                      output: 'stylesheet.css',
-                      command: [
-                        sassc, '-a', '@INPUT@', '@OUTPUT@'
-                      ],
-                      install: true,
-                      install_dir:meson.current_source_dir() )
+run_command(sassc, '-a', '_stylesheet.scss', 'stylesheet.css')
 
 extension_data += configure_file(
   input: metadata_name + '.in',
diff --git a/extensions/osk-hangul/stylesheet.css b/extensions/osk-hangul/stylesheet.css
deleted file mode 100644
index 4d47bad..0000000
--- a/extensions/osk-hangul/stylesheet.css
+++ /dev/null
@@ -1 +0,0 @@
-/* This extensions use _stylesheet.scss instead of css. Compile scss file to create stylesheet.css */
